src/lib/components/MarkdownEditor.js

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect } from "react";
+import {
+  MDXEditor,
+  codeBlockPlugin,
+  headingsPlugin,
+  listsPlugin,
+  linkPlugin,
+  quotePlugin,
+  toolbarPlugin,
+  markdownShortcutPlugin,
+  useCodeBlockEditorContext,
+} from "@mdxeditor/editor";
+import { MarkdownToolbar } from "./MarkdownToolbar.js";
+
+import "./md-editor.css";
+
+const PlainTextCodeEditorDescriptor = {
+  match: () => true,
+  priority: 0,
+  Editor: (props) => {
+    const cb = useCodeBlockEditorContext();
+    return (
+      <div onKeyDown={(e) => e.nativeEvent.stopImmediatePropagation()}>
+        <textarea
+          rows={3}
+          cols={20}
+          defaultValue={props.code}
+          onChange={(e) => cb.setCode(e.target.value)}
+        />
+      </div>
+    );
+  },
+};
+
+export const MarkdownEditor = (props) => {
+  const ref = React.useRef(null);
+  useEffect(() => {
+    if (props.selectedEmoji) {
+      const newMd = `${ref.current.getMarkdown()} ${props.selectedEmoji}`;
+      ref.current.setMarkdown(newMd);
+    }
+    if (!props.message) {
```

Comment:
else if? do you need them both to execute?

Reply:
updated , no need for both execution